### PR TITLE
BL-1106-bd-current-state-of-the-main-branch-will-not-run

### PR DESCRIPTION
**Pull request by tkozzer**
## Description

1. Created a new `application.properties` and `application-dev.properties` with more documentation.
    - The `application-dev.properties` will be the active profile and both will depend on a `secrets.properties` file
       that will need to be created upon cloning.
2. Added a new section to the `README.md` file that will walk someone through the process of running this program on their local machine.
   - Over time this will evolve and be added to. I left a `TODO` section for a windows user to fill in the proper method of running docker on a windows machine
3. I removed `application.properties` and `application-dev.properties` from `.gitignore` and added `secrets.properties`
4. Removed a dependency that was added and never used when working on auth0 implementation
5. Added a `docker-compose.yml` file.
    - This file will create a docker container with a postgres db. I marked this method as preferred, so you don't have to worry about managing postgres on your machine.
6. Deleted the `EnableWebSecurity.java` interface because it was made by mistake and added the appropriate import statement to the `SecurityConfig.java` file.
7. Commented out `allowed.origin` instance and references, since this will be used in the future but isn't relevant right now
8. Removed the `InMemoryUserRepository.java` because it won't be needed anymore

Fixes # https://bloomtechlabs.atlassian.net/browse/BL-1106

## Loom Video

[Walk through of PR](https://www.loom.com/share/7ac1ea36999246129a5e5525a5fbabbf)
Forgot to talk about adding docker, so I provided some additional info [here](https://www.loom.com/share/d8b4a92dbf064d71844b5d459bcc4e93).

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have removed unnecessary comments/console logs from my code
- [X] I have made corresponding changes to the documentation if necessary (optional)
- [X] My changes generate no new warnings
- [X] I have checked my code and corrected any misspellings
- [X] No duplicate code left within changed files
- [X] Size of pull request kept to a minimum
- [X] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by sarahlineb — APPROVED** [review]
Very organized documentation

**Review by razborovdb — APPROVED** [review]

**Review by deepalisethia — APPROVED** [review]
Looks good.

**Review by JackNunnink — APPROVED** [review]

**Review by dpearson00 — APPROVED** [review]
Great work Tyler!

**Review by AbeKalovsky — APPROVED** [review]
Nice Job cleaning up our messes Tyler. Everything looks good!

File: README.md [hunk 1]
```diff
@@ -19,3 +19,39 @@ Labs teams must follow all [Labs Engineering Standards](https://bloomtechlabs.gi
 - `awssecretkey` - authorization token for aws instance (eg. SUPERSECRET)
 
 See application-dev.properties for example values
+
```
Comment (ramib5734) [inline README.md:22]: good addition to the readme!

**Comment by Kubwayojc** [comment]
Thanks Tyler for providing clarity in your documentation and solving this issue

**Pull request merged**